src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -1006,7 +1006,7 @@ CHIP_ERROR BLEManagerImpl::ConfigureScanResponseData(ByteSpan data)
     }
     memcpy(scanResponseBuffer, data.data(), data.size());
     ByteSpan scanResponseSpan(scanResponseBuffer);
-    mScanResponse = chip::Optional(scanResponseSpan);
+    mScanResponse = chip::Optional<ByteSpan>(scanResponseSpan);
```

Comment:
For future reference, @jadhavrohit924, something like `mScanResponse = chip::MakeOptional(scanResponseSpan);` might be simpler

Reply:
@jadhavrohit924 Please check